### PR TITLE
Unify location of record types in conda/models/records.py

### PR DESCRIPTION
**Pull request by kalefranz**
Another refactor in preparation for conda 4.5.  This should be reviewed only after #6922 is merged.

### CODE REVIEW

**Comment by kalefranz** [comment]
I'll rebase this PR after we get #6922 finalized and merged.

**Comment by kalefranz** [comment]
Just had to rebase

File: conda/api.py [hunk 1]
```diff
@@ -4,11 +4,10 @@
 from .common.constants import NULL
 from .core.package_cache_data import PackageCacheData as _PackageCacheData
 from .core.prefix_data import PrefixData as _PrefixData
-from .core.solve import Solver as _Solver, DepsModifier as _DepsModifier
+from .core.solve import DepsModifier as _DepsModifier, Solver as _Solver
```
Comment (goanpeca) [inline conda/api.py]: Why do we do 

```
from .core.solve import DepsModifier as _DepsModifier
```

and then

```
DepsModifier = _DepsModifier
```

File: conda/core/package_cache_data.py [hunk 2]
```diff
@@ -28,9 +28,9 @@
                                   read_index_json_from_tarball, read_repodata_json)
 from ..gateways.disk.test import file_path_is_writable
 from ..models.dist import Dist
-from ..models.index_record import PackageRecord, PackageRef
+from ..models.records import PackageRecord, PackageRef
 from ..models.match_spec import MatchSpec
-from ..models.package_cache_record import PackageCacheRecord
+from conda.models.records import PackageCacheRecord
```
Comment (goanpeca) [inline conda/core/package_cache_data.py]: Why the mixture of import types??

`conda.models.records` vs `..models.records`

File: conda/core/path_actions.py [hunk 3]
```diff
@@ -36,10 +36,9 @@
 from ..history import History
 from ..models.channel import Channel
 from ..models.enums import LinkType, NoarchType, PathType
-from ..models.index_record import Link, PackageRecord, PathDataV1, PathsData
+from ..models.records import Link, PackageRecord, PathDataV1, PathsData
 from ..models.match_spec import MatchSpec
-from ..models.package_cache_record import PackageCacheRecord
-from ..models.prefix_record import PrefixRecord
+from conda.models.records import PackageCacheRecord, PrefixRecord
```
Comment (goanpeca) [inline conda/core/path_actions.py]: Why not join imports ?

File: conda/core/prefix_data.py [hunk 4]
```diff
@@ -16,9 +16,9 @@
 from ..gateways.disk.delete import rm_rf
 from ..gateways.disk.test import file_path_is_writable
 from ..models.dist import Dist
-from ..models.index_record import PackageRef
+from ..models.records import PackageRef
 from ..models.match_spec import MatchSpec
-from ..models.prefix_record import PrefixRecord
+from conda.models.records import PrefixRecord
```
Comment (goanpeca) [inline conda/core/prefix_data.py]: same here?

File: conda/models/dist.py [hunk 5]
```diff
@@ -21,6 +21,9 @@
                                          'dist_name'))
 
 
+IndexRecord = PackageRecord  # for conda-build backward compat
```
Comment (goanpeca) [inline conda/models/dist.py]: Is conda build accessing this module directly or should we expose it in the `api` module then?

Comment (kalefranz) [reply conda/models/dist.py]: It should be in exports right now.  That's where conda-build should be accessing it from.  I'm not sure I want to expose it yet as something that can be created through the api.  Maybe; maybe not.

File: conda/core/prefix_data.py (hunk 4 above)
Comment (kalefranz) [reply conda/core/prefix_data.py]: thank, yeah that's a problem

File: conda/api.py (hunk 1 above)
Comment (kalefranz) [reply conda/api.py]: Yeah we don't need that.

File: conda/models/dist.py (hunk 5 above)
Comment (goanpeca) [reply conda/models/dist.py]: Yeah exports module, so should we add this renaming there...?

File: conda/api.py (hunk 1 above)
Comment (kalefranz) [reply conda/api.py]: Actually, did that to avoid the reassignment error thing.

Comment (goanpeca) [reply conda/api.py]: Yeah but I mean why not import it directly as 

`from .core.solve import DepsModifier` ?

Comment (goanpeca) [reply conda/api.py]: @kalefranz?

File: conda/models/dist.py (hunk 5 above)
Comment (goanpeca) [reply conda/models/dist.py]: @kalefranz ?

File: conda/api.py (hunk 1 above)
Comment (kalefranz) [reply conda/api.py]: Because then you need a line

    DepsModifier = DepsModifier

File: conda/models/dist.py (hunk 5 above)
Comment (kalefranz) [reply conda/models/dist.py]: We can't for a while.

https://github.com/conda/conda-build/blob/3.5.1/conda_build/conda_interface.py#L67

**Comment by kalefranz** [comment]
Merging following review by @goanpeca

**Pull request merged**